loadbalancer/zk/client.go

```diff
@@ -0,0 +1,157 @@
+package zk
+
+import (
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/eapache/channels"
+	"github.com/samuel/go-zookeeper/zk"
+
+	"github.com/go-kit/kit/log"
+)
+
+// DefaultACL is the default ACL to use for creating znodes
+var DefaultACL = zk.WorldACL(zk.PermAll)
+
+const (
+	// ConnectTimeout is the default timeout to establish a connection to a
+	// ZooKeeper node.
+	ConnectTimeout = 2 * time.Second
+	// SessionTimeout is the default timeout to keep the current ZooKeeper
+	// session alive during a temporary disconnect.
+	SessionTimeout = 5 * time.Second
+)
+
+// Client is a wrapper around a lower level ZooKeeper client implementation.
+type Client interface {
+	// GetEntries should query the provided path in ZooKeeper, place a watch on
+	// it and retrieve data from its current child nodes.
+	GetEntries(path string) ([]string, channels.SimpleOutChannel, error)
+	// CreateParentNodes should try to create the path in case it does not exist
+	// yet on ZooKeeper.
+	CreateParentNodes(path string) error
+}
+
+type clientConfig struct {
+	acl             []zk.ACL
+	connectTimeout  time.Duration
+	sessionTimeout  time.Duration
+	rootNodePayload [][]byte
+}
+
+type configOption func(c *clientConfig) error
+
+type client struct {
+	*zk.Conn
+	clientConfig
+	Eventc <-chan zk.Event
```

Comment:
I need to change that to not be exported. I will add a configuration option to allow users to provide a session event handler as they might want to know about ZooKeeper cluster updates and provide a default handler just logging the event.